File: Domain/TransApp.Domain/Addresses/AddressModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using TransApp.Domain.Common.Entity;

namespace TransApp.Domain.Addresses
{
    public class AddressModel
    {

        public int Id { get; set; }

        public int? CustomerId { get; set; }

        public string Name { get; set; }

        public AddressLocationModel Location { get; set; }

        public string ContactPerson { get; set; }

        public string Email { get; set; }

        public string Phone { get; set; }

        public string Remark { get; set; }

        public List<AvailabilityEntityModel> Availabilities { get; set; }
        public List<FacilityEntityModel> Facilities { get; set; }
        public List<RequirementEntityModel> Requirements { get; set; }
        public List<TruckEntityModel> Trucks { get; set; }

        public int? UserIdCreated { get; set; }

        public string UserCreated { get; set; }

        public DateTime? DateCreated { get; set; }

        public int? UserIdModified { get; set; }

        public string UserModified { get; set; }

        public DateTime? DateModified { get; set; }

        public bool CommonAvailability { get; set; }

        public string State { get; set; }
        public string OpeningHours { get; set; }
    }
}

File: Site/Presentation/TransApp.Site.FrontEnd/src/app/address/models/address-model.ts
import { AddressLocationModel } from 'app/address/models/address-location-model';
import { FacilityEntityModel } from 'app/shared/common/models/entity/facility-entity-model';
import { AvailabilityEntityModel } from 'app/shared/common/models/entity/availability-entity-model';
import { RequirementEntityModel } from 'app/shared/common/models/entity/requirement-entity-model';
import { TruckEntityModel } from 'app/shared/common/models/entity/truck-entity-model';

export class AddressModel {
    public id: number;
    public customerId: number;
    public name: string;
    public location: AddressLocationModel;
    public contactPerson: string;
    public email: string;
    public phone: string;
    public remark: string;
    public userIdCreated: number;
    public dateCreated: Date;
    public userIdModified: number;
    public dateModified: Date;
    public userCreated: string;
    public userModified: string;
    public availabilities: Array<AvailabilityEntityModel>;
    public facilities: Array<FacilityEntityModel>;
    public requirements: Array<RequirementEntityModel>;
    public trucks: Array<TruckEntityModel>;
    public commonAvailability: boolean;
}
